### ISSUE TITLE
Links are misbehaving in Webpages

### ISSUE THREAD

**Issue by MikaalSky** [Khan/live-editor#351]
https://www.khanacademy.org/computer-programming/multitool/5818019692544000
Click on the link on the bottom in chrome, and whatever other browsers you want to test with. It either:
Does nothing
Gives you a 403 for some reason
...

**Comment by SpongeJr** [comment]
Yeah, it's never been quite right. Sorry for the lengthy post, but this all looks to be a similar issue, so I think this is all related info:

https://www.khanacademy.org/computer-programming/demonstration-of-ka_lljs/5036077007110144?qa_expand_key=ag5zfmtoYW4tYWNhZGVteXJpCxIIVXNlckRhdGEiRnVzZXJfaWRfa2V5X2h0dHA6Ly9pZC5raGFuYWNhZGVteS5vcmcvYzdmNTAwNTc2NWI0NDdmODhkYTBiYTFkZjVlZGJkMjUMCxIIRmVlZGJhY2sYgICAoLn32AgM

and

https://www.khanacademy.org/computer-programming/demonstration-of-ka_lljs/5036077007110144?qa_expand_key=kaencrypted_c047b55634856a12af207432eb531c08_16dd6e5434c8cf08bcbc01f096a0d6cd84bfb9fbd61777dff6f0f6a298a00a86018bc5d601c70b13892858f0bbc9161e6ecd5726bb4d53fb96dc289e21eddad950bedd740319dd6ddb6c7247f2f91162576b3cd71fa9d0eaab927debd052382c441cc74cdd57c504528ed536887e7fa6e0821fbdaac73aad71ae399f0675ece01c933c0241aa89fbb7d92780422b025e8b0d7ca55d53f2a18cd5c73bb22e59fe

The common thread on all 3 of these is this console message from that second Help Request, I think:
My link doesn't work. Checking my console messages, I see this:
```Refused to frame 'https://github.com/SpongeJr/KA-Learner-Library' because it violates the following Content Security Policy directive: "child-src *.kastatic.org http://www.khanacademy.org 'self'".```
and
```scratchpads-exec-webpage-package-666785.js:2361 Uncaught SecurityError: Sandbox access violation: Blocked a frame at "https://www.kasandbox.org" from accessing a frame at "https://www.kasandbox.org".  The frame being accessed is sandboxed and lacks the "allow-same-origin" flag.```


It's trying to open it inside that frame (which isn't allowed, of course) instead of using the link redirector, right?
I've come across this in other situations, like with dynamic content created by JS (as in the Computing Contest database project in the first link), but I don't get why it's happening here, it's just an ordinary <a> tag, there in the HTML.

I was thinking it was because I had invalid HTML (a bad comment syntax) in the most recent case, but that doesn't seem to be it.

**Comment by kevinb7** [comment]
Something funky is happening that's for sure.  I copy/pasted the code from the "Demonstration of KA_LL.js" example into a fresh webpage scratchpad and it worked fine.  After saving it though it stopped working.

**Comment by kevinb7** [comment]
I tried it again and pasting into a fresh webpage doesn't work either.  The link by itself does work.

**Comment by kevinb7** [comment]
I filed a bug on our internal tracker.

**Comment by SpongeJr** [comment]
Any updates on this? I keep bumping into it. It seems to have something to do with how you arrive at the page/scratchpad causing different behaviors-- one way gives you the link redirector, the other way and it tries to open them in the frame. That's about all I've got. Are there two different versions of something somewhere, like blank project scratchpads?

**Comment by kevinb7** [comment]
@SpongeJr sorry no update.  Thanks for posting your observations.  I have a hunch though.  There is a function called `postProcessing()` in webpage-output.js which rewrite `<a>` tags.  I think this might be running too early/late.  I recently dealt with another bug where we look at the DOM too early and were missing out on user code in `<script>` tags so the challenges were intermittently failing.
